engine/src/mac-dialog.mm

```diff
@@ -626,12 +626,13 @@ void MCPlatformBeginFolderOrFileDialog(MCPlatformFileDialogKind p_kind, MCPlatfo
     else
     {
         [t_panel setPrompt: @"Choose"];
-        [t_panel setCanChooseFiles: NO];
-        [t_panel setCanChooseDirectories: YES];
-        [t_panel setAllowsMultipleSelection: NO];
+        [(NSOpenPanel *)t_panel setCanChooseFiles: NO];
+        [(NSOpenPanel *)t_panel setCanChooseDirectories: YES];
+        [(NSOpenPanel *)t_panel setAllowsMultipleSelection: NO];
     }
     
-    if ([t_panel respondsToSelector:@selector(isAccessoryViewDisclosed)])
+    if (p_kind != kMCPlatformFileDialogKindSave && 
+        [t_panel respondsToSelector:@selector(isAccessoryViewDisclosed)])
     {
         // show accessory view when dialog opens
         [t_panel setAccessoryViewDisclosed: YES];
```

Comment:
This change does eliminate the eventual crash and allow the save dialog to appear, but does not eliminate the compiler warning.  I added a type cast and recompiled and still see the following:
```
/Users/milby/projects/GitHub/livecode/engine/src/mac-dialog.mm:638:33: warning: 
      instance method '-setAccessoryViewDisclosed:' not found (return type
      defaults to 'id') [-Wobjc-method-access]
        [(NSOpenPanel *)t_panel setAccessoryViewDisclosed: YES];
                                ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /Users/milby/projects/GitHub/livecode/engine/src/mac-dialog.mm:17:
In file included from /Applications/Xcode-Dev/Xcode_8_3.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk/System/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:13:
In file included from /Applications/Xcode-Dev/Xcode_8_3.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk/System/Library/Frameworks/AppKit.framework/Headers/AppKit.h:76:
/Applications/Xcode-Dev/Xcode_8_3.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSOpenPanel.h:13:12: note: 
      receiver is instance of class declared here
@interface NSOpenPanel : NSSavePanel {
           ^
2 warnings generated.
```